docs/standard/serialization/system-text-json/polymorphism.md

```diff
@@ -136,7 +139,607 @@ The following example shows the JSON that results from the preceding code:
+'     "TemperatureCelsius": 15,
+'     "Summary": "Cool"
+'   }
+```
+
+With the type discriminator, the serializer can deserialize the payload polymorphically as `WeatherForecastWithCity`:
+
+```csharp
+WeatherForecastBase value = JsonSerializer.Deserialize<WeatherForecastBase>(json);
+Console.WriteLine(value is WeatherForecastWithCity); // True
+```
+
+```vb
+Dim value As WeatherForecastBase = JsonSerializer.Deserialize(json)
+Console.WriteLine(value is WeatherForecastWithCity) // True
+```
+
+Type discriminator identifiers are valid in either `string` or `int` forms, so the following is valid:
+
+```csharp
+[JsonDerivedType(typeof(WeatherForecastWithCity), 0)]
+[JsonDerivedType(typeof(WeatherForecastWithTimeSeries), 1)]
+[JsonDerivedType(typeof(WeatherForecastWithLocalNews), 2)]
+public class WeatherForecastBase { }
+
+var json = JsonSerializer.Serialize(new WeatherForecastWithTimeSeries());
+Console.WriteLine(json);
+// Sample output:
+//   {
+//    "$type" : 1,
+//    Omitted for brevity...
+//   }
+```
+
+```vb
+<JsonDerivedType(GetType(WeatherForecastWithCity), 0)>
+<JsonDerivedType(GetType(WeatherForecastWithTimeSeries), 1)>
+<JsonDerivedType(GetType(WeatherForecastWithLocalNews), 2)>
+Public Class WeatherForecastBase
+End Class
+
+Dim json As String = JsonSerializer.Serialize(New WeatherForecastWithTimeSeries())
+Console.WriteLine(json)
+' Sample output:
+'  {
+'    "$type" : 1,
+'    Omitted for brevity...
+'  }
+```
+
+### Mix and match type discriminator configurations
+
+While the API supports mixing and matching type discriminator configurations, it is not recommended. The general recommendation is to use either all `string` type discriminators, all `int` type discriminators, or no discriminators at all. The following example shows how to mix and match type discriminator configurations:
+
+```csharp
+[JsonDerivedType(typeof(ThreeDimensionalPoint), typeDiscriminator: 3)]
+[JsonDerivedType(typeof(FourDimensionalPoint), typeDiscriminator: "4d")]
+public class BasePoint
+{
+    public int X { get; set; }
+    public int Y { get; set; }
+}
+
+public class ThreeDimensionalPoint : BasePoint
+{
+    public int Z { get; set; }
+}
+
+public sealed class FourDimensionalPoint : ThreeDimensionalPoint
+{
+    public int W { get; set; }
+}
+```
+
+```vb
+<JsonDerivedType(GetType(ThreeDimensionalPoint), 3)>
+<JsonDerivedType(GetType(FourDimensionalPoint), "4d")>
+Public Class BasePoint
+    Public Property X As Integer
+    Public Property Y As Integer
+End Class
+
+Public Class ThreeDimensionalPoint
+    Inherits BasePoint
+    Public Property Z As Integer
+End Class
+
+Public NotInheritable Class FourDimensionalPoint
+    Inherits ThreeDimensionalPoint
+    Public Property W As Integer
+End Class
+```
+
+In the preceding example, the `BasePoint` doesn't have a type discriminator, while the `ThreeDimensionalPoint` type has an `int` type discriminator, and the `FourDimensionalPoint` has a `string` type discriminator.
```

Comment:
```suggestion
In the preceding example, the `BasePoint` type doesn't have a type discriminator, while the `ThreeDimensionalPoint` type has an `int` type discriminator, and the `FourDimensionalPoint` has a `string` type discriminator.
```

Reply:
Consider explaining when to add a type discriminator for the base type (like in the previous example) and when it's not needed.